src/treelearner/feature_histogram.hpp

```diff
@@ -609,29 +670,65 @@ class FeatureHistogram {
     double sum_left_gradient = grad;
     double sum_right_gradient = sum_gradient - sum_left_gradient;
     // current split gain
-    double current_gain =
-        GetLeafGain<true, true>(sum_right_gradient, sum_right_hessian,
-                                meta_->config->lambda_l1, l2,
-                                meta_->config->max_delta_step) +
-        GetLeafGain<true, true>(sum_left_gradient, sum_left_hessian,
-                                meta_->config->lambda_l1, l2,
-                                meta_->config->max_delta_step);
+    double current_gain;
+    bool use_smoothing = meta_->config->path_smooth > kEpsilon;
+    if (use_smoothing) {
```

Comment:
there are lot of `if(use_smoothing)` in code.
```
if(use_smoothing){
    func<true, true, true>();
}else {
    func<true, true, false>();
}
```
it's better use `func<true, true, use_smoothing>();` instead

Reply:
This gives a compiler error because the value of `use_smoothing` isn't known at compile time.